plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/procedure/TestIcebergAddFilesProcedure.java

```diff
@@ -173,7 +173,62 @@ void testAddFilesTypeMismatch()
     }
 
     @Test
-    void testAddFilesDifferentDataColumnDefinitions()
+    void testAddFilesFromLessColumnTable()
+    {
+        for (String format : List.of("ORC", "PARQUET", "AVRO")) {
+            String hiveTableName = "test_add_files_" + randomNameSuffix();
+            String icebergTableName = "test_add_files_" + randomNameSuffix();
+
+            assertUpdate("CREATE TABLE hive.tpch." + hiveTableName + " WITH (format = '" + format + "') AS SELECT 1 x", 1);
+            assertUpdate("CREATE TABLE iceberg.tpch." + icebergTableName + " WITH (format = '" + format + "') AS SELECT 2 x, 20 y", 1);
```

Comment:
Is adding to target table with partition column not present in source is allowed? Could you add such test case?